be/src/storage/column_expr_predicate.cpp

```diff
@@ -285,30 +285,22 @@ Status ColumnExprPredicate::try_to_rewrite_for_zone_map_filter(starrocks::Object
 }
 Status ColumnExprPredicate::seek_inverted_index(const std::string& column_name, InvertedIndexIterator* iterator,
                                                 roaring::Roaring* row_bitmap) const {
-    // Only support like predicate for now
+    // Only support simple like predicate for now
+    // Root must be LIKE, and left child must be ColumnRef, which satisfy simple like predicate
+    // format as: col LIKE xxx, xxx can be any expr with string type
     auto* vectorized_function_call = dynamic_cast<VectorizedFunctionCallExpr*>(_expr_ctxs[0]->root());
-    if (!vectorized_function_call) {
-        int children_num = _expr_ctxs[0]->root()->get_num_children();
-        bool found_like = false;
-        for (int i = 0; i < children_num; i++) {
-            vectorized_function_call = dynamic_cast<VectorizedFunctionCallExpr*>(_expr_ctxs[0]->root()->get_child(i));
-            if (vectorized_function_call) {
-                auto function_desc = vectorized_function_call->get_function_desc();
-                if (LIKE_FN_NAME == boost::to_lower_copy(function_desc->name)) {
-                    found_like = true;
-                    break;
-                }
-            }
-        }
-        if (!found_like) {
-            return Status::NotSupported("Not supported function call in inverted index");
-        }
-    } else {
-        auto function_desc = vectorized_function_call->get_function_desc();
-        if (LIKE_FN_NAME != boost::to_lower_copy(function_desc->name)) {
-            return Status::NotSupported("Not supported predicate with seeking inverted index");
+    bool vaild_pred = false;
+    if (vectorized_function_call &&
```

Comment:
Why not use direct use function name to check?
What's `dynamic_cast` used for?

Reply:
Because we must make sure that the root node is function call and then we check the function name. For example, `upper(col) = concat("ab", "c")` is an expression predicate but the root node is not function call. It will crash if we do not check the root node.

Reply:
`dynamic_cast ` is a runtime method to check the derived type the expression. For example if the root node is a binary predicate but not function call expr, the res of dynamic_cast is nullptr.

Reply:
Fixed it, use node_type() to check the expression type instead of function name or dynamic_cast

Reply:
Updated, dynamic_cast is another runtime method to check derived type of expression. If the actual type is not function call will get nullptr.

Reply:
We should check the node first and then check the function name. Because the root node may not be a function call expression